core/geotime/src/main/java/mil/nga/giat/geowave/core/geotime/store/filter/SpatialQueryFilter.java

```diff
@@ -196,8 +204,73 @@ public void fromBinary(
 					fieldId));
 		}
 		buf.get(theRest);
-		queryGeometry = geometryInterner.intern(GeometryUtils.geometryFromBinary(geometryBinary));
+		// cannot intern the prepared geometry. It uses a Object.hashCode().
+		preparedGeometryImage = geometryImageInterner.intern(new GeometryImage(
+				geometryBinary));
+		preparedGeometryImage.init();
 
 		super.fromBinary(theRest);
 	}
+
+	/**
+	 * Used for interning Geometry. Interning a geometry provides no additional
+	 * benefit since the geometry is already constructed in the heap at the
+	 * interning begins. The byte image of geometry provides a more efficient
```

Comment:
"Interning a geometry provides no additional benefit since the geometry is already constructed in the heap at the interning begins"  -  I'm not quite sure the scope of the comment - do you mean interning geometries in general, interning geometries in addition to the byte[] value, etc?  As I read it I disagree with this comment - or at a minimum it doesn't accurately convey the issue.
So the point of interning was to have one instance of a geometry object maintained as iterator state per JVM (or I guess classloader) instance.   Since we could end up with tens of thousands of iterators per scan range a 2 MB geometry could explode into gigabytes of data, completely using up the head allocated to the tablet server.